storagemarket/deal_execution.go

```diff
@@ -585,13 +663,18 @@ func (p *Provider) indexAndAnnounce(ctx context.Context, pub event.Emitter, deal
 	// announce to the network indexer but do not fail the deal if the announcement fails
 	annCid, err := p.ip.AnnounceBoostDeal(ctx, deal)
 	if err != nil {
-		p.dealLogger.LogError(deal.DealUuid, "failed to announce deal to network indexer "+
-			"but not failing deal as it's already been handed for sealing", err)
-	} else {
-		p.dealLogger.Infow(deal.DealUuid, "announced deal to network indexer", "announcement-cid", annCid)
+		return &dealMakingError{
```

Comment:
@dirkmc I am not sure we want to be failing the deal if the announcement fails. We are changing behaviour here right now - is this intended?

Reply:
The retry type is DealRetryAuto, so the deal doesn't actually get failed. The deal is paused until either the user retries it manually, or boost automatically retries it the next time boost restarts.
Note that index and announce is the last step in the deal execution (it has already been handed off to the sealer)

Reply:
> Note that index and announce is the last step in the deal execution (it has already been handed off to the sealer)

Oh, OK, I missed that. Then I guess we are good!